PLS/Services/AgitApiClient.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using PLS.Utils;
+
+namespace PLS.Services
+{
+    public class AgitApiClient : IAgitApiClient
+    {
+        private readonly string _apiUri;
+        private readonly string _repo;
+        private readonly HttpClient _cli;
+
+        public static AgitApiClientFactory Factory { get; } = (uri, repo) => new AgitApiClient(uri, repo);
+
+        public AgitApiClient(string apiUri, string repo)
```

Comment:
Should we not make the constructor protected or private to be sure it is initialized through the Factory?